src/applications/facility-locator/actions/index.js

```diff
@@ -38,24 +48,82 @@ export function fetchVAFacility(id, facility = null) {
     return fetch(url, api.settings)
       .then(res => res.json())
       .then(
-        data => dispatch({ type: 'FETCH_VA_FACILITY', payload: data.data }),
-        err => dispatch({ type: 'SEARCH_FAILED', err })
+        data => dispatch({ type: FETCH_VA_FACILITY, payload: data.data }),
+        err => dispatch({ type: SEARCH_FAILED, err })
       );
   };
 }
 
+export const searchProviders = (bounds, serviceType, page = 1) => {
+  return (dispatch, getState) => {
+    const { searchString } = getState().searchQuery;
+    if (searchString && searchString !== '') {
+      return fetchProviders(searchString, dispatch, serviceType, page); // eslint-disable-line
+    }
+
+    reverseGeocodeBox(bounds).then(address => {
+      if (!address) {
+        dispatch({ type: SEARCH_FAILED, error: 'Reverse geocoding failed. See previous errors.' });
+        return null;
+      }
+      console.log('Reverse geocoded address:', address); //eslint-disable-line
+
+      return fetchProviders(address, dispatch, serviceType, page); // eslint-disable-line
+    });
+
+    return null;
+  };
+};
+
+const fetchProviders = (address, dispatch, serviceType, page) => {
```

Comment:
I think you could avoid passing the `dispatch` if you wrap the contents of this function in `return (dispatch) => { ... }` and do `dispatch(fetchProviders(address, serviceType, page))` whenever you need to call it.

Or if you want to continue including it, maybe have `dispatch` before all of the other params to have a clearer separation between a helper function and the actual arguments of the function.

Reply:
This, too, gets cleaned up in a later commit when I refactor:
[actions/index.js#L79](https://github.com/department-of-veterans-affairs/vets-website/blob/4c586c2342f917601ceae2b0ff171659904ac577/src/applications/facility-locator/actions/index.js#L79) & [actions/index.js#L108](https://github.com/department-of-veterans-affairs/vets-website/blob/4c586c2342f917601ceae2b0ff171659904ac577/src/applications/facility-locator/actions/index.js#L108)

But that is a good point about wrapping the calls in `dispatch` initially, it felt weird passing it anyway 😄